src/python/dxpy/nextflow/nextflow_builder.py

```diff
@@ -0,0 +1,158 @@
+            else:
+                object_id = dx_object
+                object_project = None
+            if not dxpy.utils.resolver.is_hashid(object_id):
+                object_project, _, object_id = dxpy.utils.resolver.resolve_existing_path(object_id, expected="entity", expected_classes=["file"], describe=False)
+                object_id = object_id["id"]
+            return dxpy.dxlink(object_id=object_id, project_id=object_project)
+        except dxpy.utils.resolver.ResolutionError:
+            print("GitHub credentials ('{}') file could not be found!".format(dx_object), file=sys.stderr)
+            exit(2)
+
+
+    build_project_id = dxpy.WORKSPACE_ID
+    if build_project_id is None:
+        parser.error(
+            "Can't create an applet without specifying a destination project; please use the -d/--destination flag to explicitly specify a project")
+
+    input_hash = {
+        "repository_url": repository,
+    }
+    if tag:
+        input_hash["repository_tag"] = tag
+    if profile:
+        input_hash["config_profile"] = profile
+    if github_creds:
+        input_hash["github_credentials"] = create_dxlink(github_creds)
+
+    api_options = {
+        "name": "Nextflow build of %s" % (repository),
+        "input": input_hash,
+        "project": build_project_id,
+    }
+    try:
+        app_run_result = dxpy.api.app_run('app-nextflow_pipeline_importer', input_params=api_options)
+    except dxpy.exceptions.ResourceNotFound:
+        print("GitHub credentials file ('{}') could not be found!".format(github_creds), file=sys.stderr)
+        raise
+    job_id = app_run_result["id"]
+    if not brief:
+        print("Started builder job %s" % (job_id,))
+    dxpy.DXJob(job_id).wait_on_done(interval=1)
+    applet_id, _ = dxpy.get_dxlink_ids(dxpy.api.job_describe(job_id)['output']['output_applet'])
+    if not brief:
+        print("Created Nextflow pipeline %s" % (applet_id))
+    else:
+        print(applet_id)
+    return applet_id
+
+def prepare_nextflow(resources_dir, profile):
+    """
+    :param resources_dir: Directory with all resources needed for Nextflow Pipeline. Usually directory with user's NF files.
+    :type resources_dir: str or Path
+    :param profile: Custom NF profile, for more information visit https://www.nextflow.io/docs/latest/config.html#config-profiles
+    :type profile: string
+
+    Creates files for creating applet, such as dxapp.json and source file. These files are created in temp directory.
+    """
+    assert os.path.exists(resources_dir)
+    inputs = []
+    os.makedirs(".dx.nextflow", exist_ok=True)
+    dxapp_dir = os.path.join(resources_dir, '.dx.nextflow')
+    if os.path.exists(f"{resources_dir}/nextflow_schema.json"):
+        inputs = prepare_inputs(f"{resources_dir}/nextflow_schema.json")
+    DXAPP_CONTENT = get_nextflow_dxapp(inputs)
+    EXEC_CONTENT = get_nextflow_src(inputs=inputs, profile=profile)
+    copy_tree(get_template_dir(), dxapp_dir)
```

Comment:
perhaps here we can also `copy_tree(resources_dir, get_resources_subpath())` and so the `.dx.nextflow` folder will contain
```
dxapp.json
src/nextflow.sh
resources/usr/bin/dx-registry-login
resources/home/dnanexus/ntf/{main.nf, config, etc}
```
and then the `worker_resources_subpath` could still be the default "."
@mhrvol @commandlinegirl

Reply:
The resources folder is the main folder with code. It has only one resources, therefore whatever inside resources will be ignored (or rather present at <worker_resources_subpath>/resources/...)